File: 백준/Bronze/5692. 팩토리얼 진법/팩토리얼 진법.cc
#include <stdio.h>

int factorial(int n);

int main() {
    char T[6];
    while(1) {
        scanf("%s", T);
        if(T[0] == '0' && T[1] == '\0') {
            break;
        }

        int len = 0;
        while(T[len] != '\0') {
            len++;
        }

        int res = 0;
        for(int i = 0; i < len; i++) {
            int digit = T[i] - '0';
            res += digit * factorial(len - i);
        }

        printf("%d\n", res);
    }
    return 0;
}

int factorial(int n) {
    int fact = 1;
    for(int i = 2; i <= n; i++) {
        fact *= i;
    }
    return fact;
}

File: 백준/Bronze/5692. 팩토리얼 진법/팩토리얼 진법.java
import java.util.Scanner;

public class Main {
    private static int factorial(int n) {
        int fact = 1;
        for (int i = 1; i <= n; i++) {
            fact *= i;
        }
        return fact;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            String input = scanner.next();
            if (input.equals("0")) {
                break;
            }

            int len = input.length();
            int res = 0;

            for (int i = 0; i < len; i++) {
                int digit = Integer.parseInt(Character.toString(input.charAt(i)));
                res += digit * factorial(len - i);
            }

            System.out.println(res);
        }

        scanner.close();
    }
}
